tests/microservices/Microservices.CohortExtractor.Tests/PatientRejectorTests.cs

```diff
@@ -0,0 +1,77 @@
+﻿using FAnsi;
+using FAnsi.Discovery;
+using Microservices.CohortExtractor.Execution.RequestFulfillers;
+using Moq;
+using NUnit.Framework;
+using Rdmp.Core.Curation;
+using Rdmp.Core.Curation.Data;
+using System;
+using System.Collections.Generic;
+using System.Data.Common;
+using Tests.Common;
+
+namespace Microservices.CohortExtractor.Tests
+{
+    public class PatientRejectorTests : DatabaseTests
+    {
+        private const string PatColName = "PatientID";
+
+        [TestCase(DatabaseType.MicrosoftSQLServer)]
+        [TestCase(DatabaseType.MySql)]
```

Comment:
I'm sad that I need to spin-up a set of databases to test this. Seems like we just need to test the business logic here rather than the database interaction. Is there any way we can mock this out?

Reply:
The constructor is fully concerned with database interactions.  The reject method is basically if contains X return true.  With a case for DBNull.Value.

The larger context of the component is already tested since it is an `IRejector` and there are other tests that cover whether they report properly the results.

So I think if you distil out the database interaction you pretty much get an empty class and testing `HashSet` not the actual class.

Reply:
Hmn could do an abstract base and subclass?  The IRejector interface already handles "reject these things for reason X based on the values in the column" so this change would just streamline the idea of "based on these values and this column". 

Either way I think we want that database check to ensure it is querying the DBMS correctly etc.